### PR TITLE
[MM-60854] Replace FormattedMarkdownMessage in 'webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx' with FormattedMessage

### LINKED ISSUE #28700: Replace FormattedMarkdownMessage in 'webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx' with FormattedMessage
The goal of this task is to remove all instances of “FormattedMarkdownMessage” (imported from “components/formatted_markdown_message”) in the file `webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx`. These should be replaced with the “FormattedMessage” component from the “react-intl” library.

Ensure that rich text formatting is preserved by following the implementation guidelines in the react-intl (https://formatjs.io/docs/react-intl/components#rich-text-formatting) documentation. Make sure to change the message ID after the update as the current extraction tool does not automatically detect changes in the messages. Once done, make sure to run `npm run i18n-extract` in the “webapp/channels” directory to pick up the new messages. For guidance refer to the example pull request - https://github.com/mattermost/mattermost/pull/28438

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-60854

### PR DESCRIPTION
**Pull request by RS-labhub**
#### Summary
Replaced FormattedMarkdownMessage in '../admin_console/team_channel_settings/convert_confirm_modal.tsx' with FormattedMessage

#### Ticket Link

Fixes #28700  
Jira [MM-60854](https://mattermost.atlassian.net/browse/MM-60856)

#### Release Note

```release-note
NONE
```

## Before:
image

image
<figcaption>not getting an option to convert a pvt channel to pub channel</figcaption>

## After:
image

[MM-60854]: https://mattermost.atlassian.net/browse/MM-60854?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 1]
```diff
@@ -101,9 +113,14 @@ const messages = defineMessages({
     toPrivateMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPrivateMessage',
 
-        // This eslint-disable comment can be removed once this component no longer uses FormattedMarkdownMessage
         // eslint-disable-next-line formatjs/no-multiple-whitespaces
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:116]: i think we should be able to remove this as well

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 2]
```diff
@@ -101,9 +113,14 @@ const messages = defineMessages({
...
-        defaultMessage: 'When you convert **{displayName}** to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link. Membership in a private channel is by invitation only.  \n \nAre you sure you want to convert **{displayName}** to a private channel?',
+        defaultMessage: 'When you convert {displayName} to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link.',
+    },
+    toPrivateMessageConfirmation: {
+        id: 'admin.team_channel_settings.convertConfirmModal.toPrivateMessageConfirmation',
+
+        // eslint-disable-next-line formatjs/no-multiple-whitespaces
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:122]: i think we should be able to remove this as well

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 3]
```diff
@@ -116,9 +133,14 @@ const messages = defineMessages({
     toPublicMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPublicMessage',
 
-        // This eslint-disable comment can be removed once this component no longer uses FormattedMarkdownMessage
         // eslint-disable-next-line formatjs/no-multiple-whitespaces
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:136]: i think we should be able to remove this as well

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 4]
```diff
@@ -101,9 +113,14 @@ const messages = defineMessages({
     toPrivateMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPrivateMessage',
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:114]: even this message is now changed then but the en.json isnt updated with this change

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 5]
```diff
@@ -101,9 +113,14 @@ const messages = defineMessages({
...
+        defaultMessage: 'When you convert {displayName} to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link.',
+    },
+    toPrivateMessageConfirmation: {
+        id: 'admin.team_channel_settings.convertConfirmModal.toPrivateMessageConfirmation',
+
+        // eslint-disable-next-line formatjs/no-multiple-whitespaces
+        defaultMessage: 'Are you sure you want to convert {displayName} to a private channel?',
     },
     toPrivateTitle: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPrivateTitle',
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:126]: similarly here as well this message is now changed then but the en.json isnt updated with this change

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by RS-labhub — COMMENTED** [review]
I forgot to push the new changes 🙂

Sorry, doing it rn!

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 4 above)
Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:114]: Done!

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 1 above)
Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx]: Done!

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 5 above)
Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:122]: Done!

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 3 above)
Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx]: Done!

**Review by RS-labhub — COMMENTED** [review]
Fixed all!

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 6]
```diff
@@ -100,10 +112,11 @@ const messages = defineMessages({
     },
     toPrivateMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPrivateMessage',
-
-        // This eslint-disable comment can be removed once this component no longer uses FormattedMarkdownMessage
-        // eslint-disable-next-line formatjs/no-multiple-whitespaces
-        defaultMessage: 'When you convert **{displayName}** to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link. Membership in a private channel is by invitation only.  \n \nAre you sure you want to convert **{displayName}** to a private channel?',
+        defaultMessage: 'When you convert {displayName} to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link.',
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:115]: Isn't this message missing the `Membership in a private channel is by invitation only` part? Or I am missing something?

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 7]
```diff
@@ -115,10 +128,11 @@ const messages = defineMessages({
     },
     toPublicMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPublicMessage',
-
-        // This eslint-disable comment can be removed once this component no longer uses FormattedMarkdownMessage
-        // eslint-disable-next-line formatjs/no-multiple-whitespaces
-        defaultMessage: 'When you convert **{displayName}** to a public channel, history and membership are preserved. Public channels are discoverable and can by joined by users on the system without invitation.  \n \nAre you sure you want to convert **{displayName}** to a public channel?',
+        defaultMessage: 'When you convert {displayName} to a public channel, history and membership are preserved. Public channels are discoverable and can by joined by users on the system without invitation.',
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: There is a typo (from before this PR). Do you mind correcting it?

```suggestion
        defaultMessage: 'When you convert {displayName} to a public channel, history and membership are preserved. Public channels are discoverable and can be joined by users on the system without invitation.',
```

**Review by larkox — CHANGES_REQUESTED** [review]
Two small things. Thank you for the contribution!

**Review by RS-labhub — COMMENTED** [review]
@larkox corrected it!

**Comment by larkox** [comment]
@RS-labhub Thanks for the changes! I think you need to update also the `en.json` file, though.

**Review by larkox — APPROVED** [review]
Thanks for the quick turnaround!

**Comment by RS-labhub** [comment]
> @RS-labhub Thanks for the changes! I think you need to update also the `en.json` file, though.

I did it with github directly, so I need to operate both files separately. 😓

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx [hunk 8]
```diff
@@ -115,10 +128,11 @@ const messages = defineMessages({
     },
     toPublicMessage: {
         id: 'admin.team_channel_settings.convertConfirmModal.toPublicMessage',
-
-        // This eslint-disable comment can be removed once this component no longer uses FormattedMarkdownMessage
-        // eslint-disable-next-line formatjs/no-multiple-whitespaces
-        defaultMessage: 'When you convert **{displayName}** to a public channel, history and membership are preserved. Public channels are discoverable and can by joined by users on the system without invitation.  \n \nAre you sure you want to convert **{displayName}** to a public channel?',
+        defaultMessage: 'When you convert {displayName} to a public channel, history and membership are preserved. Public channels are discoverable and can be joined by users on the system without invitation.',
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: `{displayName} ` was bold before and now its not, lets not change the structure

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: same in other places as well

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 8 above)
Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: image

It's still like that..

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: Oh i see apologies. But can you make a screenshot to visually verify

Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: @M-ZubairAhmed how to start the server??

Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: **Before**:
image

image
<figcaption>not getting an option to convert a pvt channel to pub channel</figcaption>

**After**:
image

Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: @M-ZubairAhmed check it

Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: > not getting an option to convert a pvt channel to pub channel
Don't know why!

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

File: webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx (hunk 8 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: awesome, please include these in the PR description as well for QA reference

Comment (RS-labhub) [reply webapp/channels/src/components/admin_console/team_channel_settings/convert_confirm_modal.tsx:131]: Already did! Thanks!!!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**